timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/NamedMinTimestampTracker.java

```diff
@@ -13,25 +13,33 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.palantir.atlasdb.timelock.lock;
 
-import com.palantir.atlasdb.timelock.util.LoggableIllegalStateException;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.StringLockDescriptor;
 import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
 import java.util.Optional;
 import java.util.SortedMap;
 import java.util.TreeMap;
 import java.util.UUID;
 import javax.annotation.concurrent.GuardedBy;
 
-public class ImmutableTimestampTracker {
+final class NamedMinTimestampTracker {
+    private final String timestampName;
```

Comment:
Would it hurt to have a small wrapper here?

Reply:
What do you mean by wrapper here?

Reply:
Just a simple 

```
public static final class NamedTimestamp {

private final String value;
...
}
```